numpy/lib/tests/test_io.py

```diff
@@ -8,20 +8,31 @@
 import time
 import warnings
 import gc
-from io import BytesIO
+import io
+from io import BytesIO, StringIO
 from datetime import datetime
+import locale
+import re
 
 import numpy as np
 import numpy.ma as ma
 from numpy.lib._iotools import ConverterError, ConversionWarning
 from numpy.compat import asbytes, bytes, unicode, Path
 from numpy.ma.testutils import assert_equal
 from numpy.testing import (
-    run_module_suite, assert_warns, assert_, assert_raises_regex,
-    assert_raises, assert_allclose, assert_array_equal, temppath, dec, IS_PYPY,
-    suppress_warnings
+    run_module_suite, assert_warns, assert_,
+    assert_raises_regex, assert_raises, assert_allclose,
+    assert_array_equal, temppath, tempdir, dec, IS_PYPY, suppress_warnings
 )
 
+def can_encode(v):
+    """ check if bytes can be decoded with default encoding """
+    try:
+        v.encode(locale.getpreferredencoding())
+        return False # no skipping
+    except UnicodeEncodeError:
+        return True
```

Comment:
I am super confused by this function:

* Comment says `decode`
* Input is `unicode` not bytes
* Function returns `False` if it can decode

Reply:
* Comment says `default encoding`, but code doesn't use `sys.getdefaultencoding()`

Reply:
Removed. It was only used to determine if `test_utf8_file_nodtype_unicode` could be run. The test has been modified.